libraries/reconstruction/resolution_monotomo.cpp

```diff
@@ -415,45 +451,46 @@ void ProgMonoTomo::amplitudeMonogenicSignal3D(MultidimArray< std::complex<double
 }
 
 
-void ProgMonoTomo::localNoise(MultidimArray<double> &noiseMap, Matrix2D<double> &noiseMatrix, int boxsize)
+
+
+void ProgMonoTomo::localNoise(MultidimArray<double> &noiseMap, Matrix2D<double> &noiseMatrix, int boxsize, Matrix2D<double> &thresholdMatrix)
 {
 //	std::cout << "Analyzing local noise" << std::endl;
 
 	//TODO: check this function
-	double kk = 0;
-	int idx_i, idx_j;
-
 	int xdim = XSIZE(noiseMap);
 	int ydim = YSIZE(noiseMap);
 
 
 	int Nx = xdim/boxsize;
 	int Ny = ydim/boxsize;
 
-	STARTINGX(noiseMap);
+	noiseMatrix.initZeros(Ny, Nx);
 
-	noiseMatrix.initZeros(Nx, Ny);
+	// For the spline regression
+	int lX=std::min(8,Nx-2), lY=std::min(8,Ny-2);
+    WeightedLeastSquaresHelper helper;
+    helper.A.initZeros(Nx*Ny,lX*lY);
+    helper.b.initZeros(Nx*Ny);
+    helper.w.initZeros(Nx*Ny);
+    helper.w.initConstant(1);
+    double hX = xdim / (double)(lX-3);
+    double hY = ydim / (double)(lY-3);
 
 	if ( (xdim<boxsize) || (ydim<boxsize) )
 		std::cout << "Error: The tomogram in x-direction or y-direction is too small" << std::endl;
 
 
-	int step = 47;
-	std::vector<double> noiseVector;
+//	int step = 47;
+	std::vector<double> noiseVector(1);
```

Comment:
Why do you construct vector of size 1? You can actually compute the number or elements you will need